performance/src/test/com/clickhouse/benchmark/clients/BenchmarkBase.java

```diff
@@ -0,0 +1,104 @@
+
+    @State(Scope.Benchmark)
+    public static class DataState {
+        @Param({"simple"})
+        String datasetSourceName;
+        ClickHouseFormat format = ClickHouseFormat.JSONEachRow;
+
+        DataSet dataSet;
+    }
+
+    public void setup(DataState dataState) throws Exception {
+        if ("simple".equals(dataState.datasetSourceName) && dataState.dataSet == null) {
+            dataState.datasetSourceName = "simple";
+            dataState.dataSet = new SimpleDataSet();
+        } else if (dataState.datasetSourceName.startsWith("file://")) {
+
+            dataState.dataSet = new FileDataSet(dataState.datasetSourceName.substring("file://".length()));
+            dataState.datasetSourceName = dataState.dataSet.getName();
+        }
+
+        LOGGER.debug("BenchmarkBase setup(). Data source " + dataState.datasetSourceName);
+        BaseIntegrationTest.setupClickHouseContainer();
+        runQuery("CREATE DATABASE IF NOT EXISTS " + DB_NAME, false);
+        DataSets.initializeTables(dataState.dataSet);
+    }
+
+    public void tearDown() {
+        runQuery("DROP DATABASE IF EXISTS " + DB_NAME, false);
+        BaseIntegrationTest.teardownClickHouseContainer();
+    }
+
+    //Connection parameters
+    public static String getPassword() {
+        return ClickHouseServerForTest.getPassword();
+    }
+    public static String getUsername() {
+        return "default";
+    }
+    public static ClickHouseNode getServer() {
+        return ClickHouseServerForTest.getClickHouseNode(ClickHouseProtocol.HTTP, isCloud(), ClickHouseNode.builder().build());
+    }
+    public static void notNull(Object obj) {
+        if (obj == null) {
+            throw new IllegalStateException("Null value");
+        }
+    }
+
+    public static void runQuery(String query, boolean useDatabase) {
+        ClickHouseNode node = getServer();
+        try (Client client = new Client.Builder()
+                .addEndpoint(Protocol.HTTP, node.getHost(), node.getPort(), isCloud())
+                .setUsername(getUsername())
+                .setPassword(getPassword())
+                .compressClientRequest(true)
+                .setDefaultDatabase(useDatabase ? DB_NAME : "default")
+                .build()) {
+            client.queryAll(query);
+        }
+    }
+
+    public static void insertData(String tableName, InputStream dataStream, ClickHouseFormat format) {
+        ClickHouseNode node = getServer();
+        try (Client client = new Client.Builder()
+                .addEndpoint(Protocol.HTTP, node.getHost(), node.getPort(), isCloud())
+                .setUsername(getUsername())
+                .setPassword(getPassword())
+                .compressClientRequest(true)
+                .setDefaultDatabase(DB_NAME)
+                .build();
+             InsertResponse response = client.insert(tableName, dataStream, format).get()) {
+            LOGGER.info("Rows inserted: {}", response.getWrittenRows());
```

Comment:
we need to validate the data is written correctly. at least a simple sanity check for a single row. can be done in a follow-up